server/server.go

```diff
@@ -249,6 +249,30 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 	}
 
 	if s.cfg.Socket != "" {
+
+		// Try to cleanup the socket file
+		sockStat, err := os.Stat(s.cfg.Socket)
+		if err == nil {
+			if sockStat.Mode().Type() != os.ModeSocket {
+				logutil.BgLogger().Warn("Unix socket has the wrong file type, not removing it",
+					zap.String("socket", s.cfg.Socket), zap.String("file_mode", sockStat.Mode().String()))
```

Comment:
Why not return an error here? The server will fail to bind on line 276, but it could be more descriptive for debugging if you bailout right here.

Reply:
Yes, you can use `fmt.Errorf` instead.

I realize it will still error successfully later on, but users when inspecting will often look at the last error. My point is that failing at the point you know you won't be able to continue is useful (and the error is more descriptive).